### PR TITLE
Derivative 1D

### PR DESCRIPTION
**Pull request by Mayank447**
Split of the test branch as discussed in #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,5 @@
-# Pseudo-Spectral-Discretization
\ No newline at end of file
+# Pseudo-Spectral-Discretization
+
+# DiracOperator Class
+
+The `DiracOperator` class implements the Dirac operator with methods for applying it to spectral coefficients, computing eigenvalues and eigenfunctions, generating a lattice structure, and transforming values between time and frequency spaces.
```
Comment (chillenzer) [inline README.md:5]:
```suggestion
The `DiracOperator` class implements the Dirac operator with methods for applying it to vectors in real-space and spectral representation. It takes a spectrum as defined below.

# Spectrum

Spectra are defined as classes which provide, among other things, knowledge about the eigenvalues and the transformations between real and spectral space.
```

File: pseudospectral/dirac_operator.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        """
+
+        if (input_space in ['real', 'spectral']) and (output_space in ['real', 'spectral']):
+            temp_1 = self.spectrum.transform(coefficients, input_space, 'spectral')
+            temp_2 = self.spectrum.eigenvalues * temp_1
+            temp_3 = self.spectrum.transform(temp_2, 'spectral', output_space)
+            return temp_3
+        
+        else:
+            raise ValueError("Unsupported space transformation.")
```
Comment (chillenzer) [inline pseudospectral/dirac_operator.py:30]:
```suggestion
            raise ValueError(f"Unsupported space transformation from {input_space} to {output_space}.")
```

File: pseudospectral/spectra/derivative_1D.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        """
+        return lambda x: np.exp(self.eigenvalues[index] * x)
+
+
+    def _eigenvalues(self):
+        """
+        Private function to return the eigenvalues of the 1D derivative operator 
+        i.e. ik for the k-th eigenfunction exp(ikx) and k = 2*pi*m/L
+        """
+        return 1j * 2 * np.pi * np.arange(self.num_lattice_points) / self.L
```
Comment (chillenzer) [inline pseudospectral/spectra/derivative_1D.py:32]: For future reference: A future PR should address #6.

File: pseudospectral/spectra/derivative_1D.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        # Perform the discrete Fast Fourier transform to go from real to spectral space
+        elif input_space == 'real' and output_space == 'spectral':
+            return scipy.fft.fft(input_vector) / self.num_lattice_points
+        
+        # Perform the inverse discrete Fast Fourier transform to go from spectral to real space
+        elif input_space == 'spectral' and output_space == 'real':
+            return scipy.fft.ifft(input_vector) * self.num_lattice_points
+        
+        else:
+            raise ValueError("Unsupported space transformation.")
```
Comment (chillenzer) [inline pseudospectral/spectra/derivative_1D.py:55]:
```suggestion
            raise ValueError(f"Unsupported space transformation from {input_space} to {output_space}.")
```

**Review by chillenzer — CHANGES_REQUESTED** [review]
Looks good! Some very minor last things.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "numpy==2.0.0",
+    "scipy==1.14.0",
+    "matplotlib==3.9.0",
+]
+
+
+[project.optional-dependencies]
+dev = [
+    "ruff==0.5.0", 
+    "pre-commit==4.6.0"
```
Comment (chillenzer) [inline pyproject.toml:24]: Not yet, either add your `.pre-commit-config.yaml` or remove this line.

Comment (Mayank447) [reply pyproject.toml]: Done

**Review by chillenzer — APPROVED** [review]

**Pull request merged**